### PR TITLE
update chaos object with retryOnConflict

### PR DESCRIPTION
**Pull request by devincd**
### What problem does this PR solve?

update chaos object with retryOnConflict . Prevent update failure due to metadata.resourceVersion change

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by WangXiangUSTC — DISMISSED** [review]
LGTM

and could you please provide the error log or issue of `updating failure due to metadata.resourceVersion change`?

**Comment by YangKeao** [comment]
@WangXiangUSTC @devincd Will retry resolve the conflict? See #791

**Comment by devincd** [comment]
> LGTM
> 
> and could you please provide the error log or issue of `updating failure due to metadata.resourceVersion change`?

Hi, @WangXiangUSTC , sorry, I can not provide because I found the error yesterday, the log didn't save and chaos controller manager has restarted.

File: controllers/twophase/types.go [hunk 1]
```diff
@@ -260,7 +260,11 @@ func updateFailedMessage(
 ) {
 	status := chaos.GetStatus()
 	status.FailedMessage = err
-	if err := r.Update(ctx, chaos); err != nil {
-		r.Log.Error(err, "unable to update chaos status")
+
+	updateError := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		return r.Update(ctx, chaos)
```
Comment (YangKeao) [inline controllers/twophase/types.go:265]: If it conflicts at first, the retry will also result in conflict. Is it meaningful to simply retry and update here?

Comment (devincd) [reply controllers/twophase/types.go:265]: Yes, You are right! I will revise and resubmit

Comment (devincd) [reply controllers/twophase/types.go:280]: Done.

**Review by STRRL — APPROVED** [review]
LGTM

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**